File: src/main/java/tech/zolhungaj/amqapi/servercommands/social/FriendRequestResponse.java
package tech.zolhungaj.amqapi.servercommands.social;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;

@CommandType("friend request")
public record FriendRequestResponse (
        @Json(name = "name") String playerName,
        String result,
        String reason
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/social/FriendRequestResponse.kt
package tech.zolhungaj.amqapi.servercommands.social

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType

@JvmRecord
@CommandType("friend request")
data class FriendRequestResponse(
    @Json(name = "name")
    val playerName: String,
    val result: Result,
    val reason: String?
) {
    enum class Result {
        @Json(name = "succes")//sic
        SUCCESS,

        @Json(name = "error")
        ERROR
    }
}
